src/sentry/shared_integrations/client.py

```diff
@@ -171,6 +151,29 @@ def track_response_data(self, code, span, error=None, resp=None):
         extra.update(getattr(self, "logging_context", None) or {})
         self.logger.info(u"%s.http_response" % (self.integration_type), extra=extra)
 
+
+class BaseApiClient(TrackResponseMixin):
+    base_url = None
+
+    allow_text = False
+
+    allow_redirects = None
+
+    integration_type = None
+
+    log_path = None
+
+    datadog_prefix = None
+
+    cache_time = 900
+
+    def __init__(self, verify_ssl=True, logging_context=None):
+        self.verify_ssl = verify_ssl
+        self.logging_context = logging_context
+
+    def get_cache_prefix(self):
```

Comment:
@NisanthanNanthakumar can we leave this in the mixin?

Reply:
@scefali we don't use that method in the `BaseInternalApiClient`. It's only used by the `BaseApiClient`. `get_cache_prefix`, `get_cached` and `head_cached` are only used by our first-party integrations. I don't think we should put those methods in the mixin bc they are not needed to for the `track_response_data` method.